### PR TITLE
feat(vault-csi-provider): init

### PR DESCRIPTION
**Pull request by eyenx**
Initialization of vault-csi-provider chart

### CODE REVIEW

**Comment by eyenx** [comment]
CI is failing because of CRD already being installed

```
namespace/vault-csi-provider-auy44i01de created
Error: rendered manifests contain a resource that already exists. Unable to continue with install: CustomResourceDefinition "secretproviderclasses.secrets-store.csi.x-k8s.io" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "vault-csi-provider-auy44i01de"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "vault-csi-provider-auy44i01de"
```

**Comment by eyenx** [comment]
rebased because of linting

**Comment by eyenx** [comment]
```
Error: rendered manifests contain a resource that already exists. Unable to continue with install: CustomResourceDefinition "secretproviderclasses.secrets-store.csi.x-k8s.io" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "vault-csi-provider-l3j9wvsqe8"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "vault-csi-provider-l3j9wvsqe8"
```

**Comment by eyenx** [comment]
Any Idea how to fix the CI in this case? @hairmare

**Comment by hairmare** [comment]
> Any Idea how to fix the CI in this case? @hairmare

hmmm..... maybe we could put the changed charts in a env here:

https://github.com/adfinis-sygroup/helm-charts/blob/10811b9a64b12d51200cef7efd804031d9ee98e5/.github/workflows/test.yaml#L48-L54

and then skip installing the CRD in the init.sh script if the env var contains the chart.

https://github.com/adfinis-sygroup/helm-charts/blob/10811b9a64b12d51200cef7efd804031d9ee98e5/hack/deploy/init.sh#L24

This wont work if we change multiple charts and the change contains a chart that needs the CRD at the same time as not needing it.

Another fix would be to amend this chart so installing the CRD is optional and to just disable the CRD in the charts `ci/` folder.

File: charts/vault-csi-provider/values.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  pullPolicy: IfNotPresent
+
+# resources -- Set the limits and requests on vault csi-driver-provider pod resources
+resources: {}
+
+# csiProviderPath -- set the path where the secrets-store-csi-provider gets installed on the node
+csiProviderPath: "/etc/kubernetes/secrets-store-csi-providers"
+
+secretsStoreCsiDriver:
+  # secretsStoreCsiDriver.enabled -- specifies wether or not the secrets-store-csi-driver dependency gets installed
```
Comment (hairmare) [inline charts/vault-csi-provider/values.yaml:19]:
```suggestion
  # secretsStoreCsiDriver.enabled -- specifies whether or not the secrets-store-csi-driver dependency gets installed
```
Could also be something like "install the secrets-store-csi-drive if true" or similar.

File: charts/vault-csi-provider/values.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  tag: 0.0.6
+  pullPolicy: IfNotPresent
+
+# resources -- Set the limits and requests on vault csi-driver-provider pod resources
+resources: {}
+
+# csiProviderPath -- set the path where the secrets-store-csi-provider gets installed on the node
+csiProviderPath: "/etc/kubernetes/secrets-store-csi-providers"
+
+secretsStoreCsiDriver:
```
Comment (hairmare) [inline charts/vault-csi-provider/values.yaml:18]: Should we add a hint to configuring the dependency chart?
```suggestion
# secretsStoreCsiDriver -- Configuration for [secrets-store-csi-driver chart](https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/master/charts)
# @default - - `{}`
secretsStoreCsiDriver:
```

**Review by hairmare — APPROVED** [review]
LGTM with 2 smol docs nitpicks

**Comment by hairmare** [comment]
@eyenx did we ever end up pinging upstream regarding them hosting their own chart?

**Pull request merged**